lib/Input.js

```diff
@@ -37,7 +47,9 @@ class Input extends React.Component {
   }
 
   componentDidUpdate (prevProps) {
-    if (this.props.autoresize && prevProps.query !== this.props.query) {
+    if (prevProps.autoresize !== this.props.autoresize ||
+      (this.props.autoresize && RESIZE_PROPS.some((prop) => this.props[prop] !== prevProps[prop]))
+    ) {
```

Comment:
Good point... How about if this condition is changed specifically for the case you describe?

```
componentDidUpdate (prevProps, prevState) {
  if (this.state.inputWidth === prevState.inputWidth) {
    this.updateInputWidth()
  }
}
```

My feeling is that there are many props and we want to trigger a re-render when all but one changes. Conversely there is only one state property for which we _don't_ want to re-render for the reasons you describe above.

Am I making any sense?